endpoint.go

```diff
@@ -44,6 +44,10 @@ func Parse(cfg string) (p Endpoint, err error) {
 		// Handle file paths that contains ":" (often happens on windows platform)
 		//
 		// See issue: https://github.com/beyondstorage/go-endpoint/issues/3
+		s[1] = strings.TrimLeft(s[1], "/")
```

Comment:
Compared to handle them in different protocol cases, how about handling all of them? Maybe we can operate on `s[1]` directly?